kustomize/commands/edit/set/setsecret_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// The duplicate linter is reporting that setconfigmap_test.go and this file are duplicates, which is not true.
+// Disabling lint for these two files specifically to work around that.
+//
+//nolint:dupl
+package set
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	testutils_test "sigs.k8s.io/kustomize/kustomize/v5/commands/internal/configmapsecret"
+)
+
+func TestFailureCasesEditSetSecret(t *testing.T) {
+	testCases := []testutils_test.FailureCase{
+		{
+			Name: "fails to set a value because key doesn't exist",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - key1=val1
+  name: test-secret
+  type: Opaque
+- literals:
+  - key3=val1
+  name: test-secret-2
+  namespace: test-ns
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--from-literal=key3=val2"},
+			ExpectedErrorMsg: "key 'key3' not found in resource",
+		},
+		{
+			Name: "fails to set a value because secret doesn't exist",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - key1=val1
+  name: test-secret
+  type: Opaque
+- literals:
+  - key2=value
+  name: another-secret
+  namespace: a-namespace
+  type: Opaque
+`,
+			Args:             []string{"test-secret2", "--from-literal=key3=val2"},
+			ExpectedErrorMsg: "unable to find Secret with name '\"test-secret2\"'",
+		},
+		{
+			Name: "fails validation because no attributes are being changed",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - a-test-key=a-test-value
+  name: test-secret
+  namespace: test-ns
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--namespace=test-ns"},
+			ExpectedErrorMsg: "at least one of [--from-literal, --new-namespace] must be specified",
+		},
+		{
+			Name: "fails when a literal source doesn't have a key",
+			KustomizationFileContent: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- literals:
+  - some-key=some-value
+  name: test-secret
+  type: Opaque
+`,
+			Args:             []string{"test-secret", "--from-literal=value"},
+			ExpectedErrorMsg: "literal values must be specified in the key=value format",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			kustomization, err := testutils_test.SetupEditSetConfigMapSecretTest(t, newCmdSetSecret, tc.KustomizationFileContent, tc.Args)
+
+			require.Nil(t, kustomization)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tc.ExpectedErrorMsg)
+		})
+	}
+}
+
+func TestSuccessCasesEditSetSecret(t *testing.T) {
```

Comment:
Could you add test cases when the 'kustomization.yaml' file already has contents?
e.g., The kustomization.yaml already has a secretGenerator field.

Reply:
All the test cases here will create a `kustomization.yaml` with one existing `secretGenerator` -- `KustomizationFileContent` is the content with which the `kustomization.yaml` will be created before running the command. Did you mean a case where that field is empty?

Reply:
Ah, thanks. I understand those tests do.
Yes, I think we need to add any tests when `kustomization.yaml` is empty.

Reply:
Added the new tests in ce316b4 and a quoting fix in aaff543. Let me know your thoughts! 😃

Reply:
Thanks!